OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs

```diff
@@ -92,8 +92,8 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 
 			foreach (var kvp in palettes)
 			{
-				if ((info.Palettes.Count > 0 && !info.Palettes.Any(kvp.Key.StartsWith))
-					|| (info.ExcludePalettes.Count > 0 && info.ExcludePalettes.Any(kvp.Key.StartsWith)))
+				if ((info.Palettes.Count > 0 && !AnyPaletteNameStartsWith(info.Palettes, kvp.Key))
```

Comment:
We set up the palettes during map load, and then they don't change during the game.  Having to do N string comparisons for every frame we draw to the screen sucks.

Out of scope for this PR, but it would be nice if someone could redesign `IPaletteModifier` to avoid this.  Even just caching a mapping from string name -> integer ID (like we did for terrain types) should be a win.